kernel/rtlil.cc

```diff
@@ -767,7 +761,14 @@ void RTLIL::Module::makeblackbox()
 		delete it->second;
 	processes.clear();
 
+	connections_.clear();
+
+	pool<RTLIL::Wire*> delwires;
+	for (auto it = wires_.begin(); it != wires_.end(); ++it)
+		if (!it->second->port_input && !it->second->port_output)
+			delwires.insert(it->second);
 	remove(delwires);
```

Comment:
Uh, it seems that wires were already removed last, it's just that the list of wires to delete was *collected* before (which shouldn't matter performance-wise).

Reply:
Ah, that's right. I misread it. Clearing connections fix still stands.